### ISSUE TITLE
The update process has encountered an error: Git error: Error 129 git not found: Please update manually!

### ISSUE THREAD

**Issue by nicfab** [FreshRSS/FreshRSS#6512]
### Describe the bug

A few days ago, I stumbled upon this issue.
From the backend of FreshRSS => Settings => Update, I see an update as in the image below:

![sshot_2024-05-29_19 12 45](https://github.com/FreshRSS/FreshRSS/assets/24752671/980315c7-5da5-4c2a-9942-6e0d722d740f)

After clicking on the button "Update," I receive the error message as in the image below:

![sshot_2024-05-29_19 12 58](https://github.com/FreshRSS/FreshRSS/assets/24752671/eef1f5c2-3521-4ec9-8492-6d8bcc33180e)

Specifically, the error message says, "_The update process has encountered an error: Git error: Error 129 git not found: Please update manually!_"

### To Reproduce

On the server side, I go under `/usr/share/FreshRSS` and run `git pull`, and the response is `Already up to date.`

### Expected behavior

I expect the update from the backend. 

### FreshRSS version

1.24.0

### Environment information

- Database version: mysql  Ver 15.1 Distrib 10.6.16-MariaDB, for debian-linux-gnu (x86_64) using  EditLine wrapper
- PHP version: PHP 8.2.19 (cli) (built: May 23 2024 12:34:42) (NTS)
- Installation type: manual install
-Web server type: nginx
- OS: Ubuntu 22.04.4 LTS
- Browser: Firefox

**Comment by Alkarex** [comment]
Hello,
Have you been able to update using the same method before?

My first guess is that the problem is due to a wrong PATH, maybe due to nginx.

Check the location of your git command line:

```console
$ which git
/usr/bin/git
```

Then try to put a new file `./FreshRSS/p/phpinfo.php` with:

```php
<?php
phpinfo();
```

and access it from your Web browser with `https://freshrss.example.net/phpinfo.php` then search for the `PATH` information:
![image](https://github.com/FreshRSS/FreshRSS/assets/1008324/39042507-37f1-4404-9cc6-f77308b4883f)

**Comment by nicfab** [comment]
I did as you instructions.
I read: `PATH 	/usr/local/bin:/usr/bin:/bin `
I underline that after the update to the latest version it worked fine.

**Comment by Alkarex** [comment]
Have you been able to update using the same method before?

**Comment by nicfab** [comment]
> Have you been able to update using the same method before?

Yes, as I said, after the latest update to version 1.24.0 it worked.

**Comment by Alkarex** [comment]
To reproduce the fact that there has some new git commits, please try to go back by one commit:

```sh
git reset --hard 9de4f23bb72c9880bfc69f84f48aeb6b0215af54
chown www-data:www-data -R .
```

Then please try to comment out the following four lines:

https://github.com/FreshRSS/FreshRSS/blob/848c8195ddfbe72c7d87a7c151a8db4356946bd8/app/Controllers/updateController.php#L23-L26

And try again to update

**Comment by nicfab** [comment]
> exec

It worked.
What was the issue?

**Comment by Alkarex** [comment]
Could you please try again without commenting out the lines?

```sh
git reset --hard 9de4f23bb72c9880bfc69f84f48aeb6b0215af54
chown www-data:www-data -R .
```

And then try to update via the Web interface

**Comment by nicfab** [comment]
> Could you please try again without commenting out the lines?
> 
> ```shell
> git reset --hard 9de4f23bb72c9880bfc69f84f48aeb6b0215af54
> chown www-data:www-data -R .
> ```
> 
> And then try to update via the Web interface

I received the same error before: `The update process has encountered an error: Git error: Error 129 git not found: Please update manually!`

**Comment by Alkarex** [comment]
Thanks for the tests. Strange that I cannot reproduce the issue with an Ubuntu test.

What does the following command return?

```sh
git -v
```

**Comment by nicfab** [comment]
> Thanks for the tests. Strange that I cannot reproduce the issue with an Ubuntu test.
> 
> What does the following command return?
> 
> ```shell
> git -v
> ```

If you meant to retrieve the version, the correct command should be `git --version`.
Was it so?

**Comment by nicfab** [comment]
> > Thanks for the tests. Strange that I cannot reproduce the issue with an Ubuntu test.
> > What does the following command return?
> > ```shell
> > git -v
> > ```
> 
> If you meant to retrieve the version, the correct command should be `git --version`. Was it so?

In that case, `git version 2.34.1`

**Comment by Frenzie** [comment]
-v also works on 2.43.0, in that case the cause of the problem has probably been found. ;-)

**Comment by Alkarex** [comment]
Indeed, I have just realised that `git -v` is only available since git 2.37 https://git-scm.com/docs/git/2.37.0 , which means Ubuntu 23.10+
Thanks for the debugging help. Patch coming

**Comment by nicfab** [comment]
> Indeed, I have just realised that `git -v` is only available since git 2.37 https://git-scm.com/docs/git/2.37.0 , which means Ubuntu 23.10+ Thanks for the debugging help. Patch coming

How will I be able to update?
Now it doesn't work.

**Comment by Alkarex** [comment]
https://github.com/FreshRSS/FreshRSS/pull/6513

When the PR above is merged (edit: done, go ahead), please perform again https://github.com/FreshRSS/FreshRSS/issues/6512#issuecomment-2138198153 and it should be back to a working state
